src/solvers/dg_curved/dg_2d.jl

```diff
@@ -79,8 +101,11 @@ end
 
 @inline function calc_interface_flux!(surface_flux_values, left_element, right_element, 
                                       orientation, u, equations, dg::DG, cache)
+  if left_element <= 0 # left_element = -1 at boundaries
+    return surface_flux_values
+  end
```

Comment:
I tested it and got some weird results:

Linear scalar advection (fully periodic):
```
With if-condition
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     35.900 μs (0.00% GC)
  median time:      36.900 μs (0.00% GC)
  mean time:        37.303 μs (0.00% GC)
  maximum time:     71.200 μs (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     1

Without if-condition
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     23.800 μs (0.00% GC)
  median time:      24.200 μs (0.00% GC)
  mean time:        24.331 μs (0.00% GC)
  maximum time:     48.400 μs (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     1
```
Removing the if-condition makes this a lot faster, as expected.

Compressible Euler (also fully periodic):
```
With if-condition
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     73.100 μs (0.00% GC)
  median time:      74.100 μs (0.00% GC)
  mean time:        74.499 μs (0.00% GC)
  maximum time:     108.600 μs (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     1

Without if-condition:
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     75.800 μs (0.00% GC)
  median time:      76.600 μs (0.00% GC)
  mean time:        76.916 μs (0.00% GC)
  maximum time:     100.100 μs (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     1
```
Here exactly the opposite happens. I retested this several times and made sure that the true-branch is never executed, so adding the condition doesn't exit the function earlier. This is `elixir_euler_source_terms.jl` in 2D.

A non-periodic example makes this even faster for Euler (which is of course expected again, since we skip some rotated flux computations):
```
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     68.100 μs (0.00% GC)
  median time:      69.000 μs (0.00% GC)
  mean time:        69.315 μs (0.00% GC)
  maximum time:     99.200 μs (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     1
```

Reply:
Which benchmark code did you run?

Reply:
```julia
using Trixi
using BenchmarkTools
trixi_include("examples/2d/elixir_euler_source_terms_curved.jl")

u = Trixi.wrap_array(sol.u[begin], semi)

@benchmark Trixi.calc_interface_flux!($u, $mesh, $equations, $solver, $semi.cache)
```

Reply:
Can you please add an extended comment to this if condition that explains that this has some performance issues but that they were not as large as to require an immediate optimization process. Please also create an issue that mentions that this is to be investigated in the future, and link to the issue in the comment. Than you can resolve this conversation here.